osm2pgsql.lua

```diff
@@ -499,130 +499,131 @@ function osm2pgsql.process_node(object)
+        output_hstore['place'] = 'country'
+    end
+-- Christmas Island
+    if object.tags.place == 'territory' and object.tags['ISO3166-1'] == 'CX' then
+        output_hstore['place'] = 'country'
+    end
+-- Cocos (Keeling) Islands
+    if object.tags.place == 'territory' and object.tags['ISO3166-1'] == 'CC' then
+        output_hstore['place'] = 'country'
+    end
+-- French Polynesia
+    if object.tags.place == 'state' and object.tags['ISO3166-1'] == 'PF' then
+        output_hstore['place'] = 'country'
+    end
+-- French Southern and Antarctic Lands
+    if object.tags.place == 'state' and object.tags['ISO3166-1'] == 'TF' then
+        output_hstore['place'] = 'country'
+    end
+-- Guam
+    if object.tags.place == 'state' and object.tags['ISO3166-1'] == 'GU' then
+        output_hstore['place'] = 'country'
+    end
+-- Heard Island and McDonald Islands
+    if object.tags.place == 'territory' and object.tags['ISO3166-1'] == 'HM' then
+        output_hstore['place'] = 'country'
+    end
+-- New Caledonia
+    if object.tags.place == 'archipelago' and object.tags['ISO3166-1'] == 'NF' then
+        output_hstore['place'] = 'region'
+    end
+-- Norfolk Island
+    if object.tags.place == 'territory' and object.tags['ISO3166-1'] == 'NF' then
+        output_hstore['place'] = 'country'
+    end
+-- Northern Mariana Islands
+    if object.tags.place == 'state' and object.tags['ISO3166-1'] == 'MP' then
+        output_hstore['place'] = 'country'
+    end
+-- Puerto Rico
+    if object.tags.place == 'state' and object.tags['ISO3166-1'] == 'PR' then
+        output_hstore['place'] = 'country'
+    end
+-- Saint Barthelemy
+    if object.tags.place == 'state' and object.tags['ISO3166-1'] == 'BL' then
+        output_hstore['place'] = 'country'
+    end
+-- Saint Helena, Ascension and Tristan da Cunha
+    if object.tags.place == 'state' and object.tags['ISO3166-1'] == 'SH' then
+        output_hstore['place'] = 'country'
+    end
+-- Saint Martin
+    if object.tags.place == 'state' and object.tags['ISO3166-1'] == 'MF' then
+        output_hstore['place'] = 'country'
+    end
+-- Saint Pierre and Miquelon
+    if object.tags.place == 'state' and object.tags['ISO3166-1'] == 'PM' then
+        output_hstore['place'] = 'country'
+    end
+-- Svalbard
+    if object.tags.place == 'region' and object.tags['ISO3166-1'] == 'SJ' then
+        output_hstore['place'] = 'country'
+    end
+-- United States Virgin Islands
+    if object.tags.place == 'state' and object.tags['ISO3166-1'] == 'VI' then
+        output_hstore['place'] = 'country'
+    end
+-- Wallis and Futuna
+    if object.tags.place == 'state' and object.tags['ISO3166-1'] == 'WF' then
+        output_hstore['place'] = 'country'
+    end
+
+-- Recast various French overseas departments as region
+-- Réunion
+    if object.tags.place == 'state' and object.tags['ISO3166-1'] == 'RE' then
+        output_hstore['place'] = 'region'
```

Comment:
This will result in no label drawn a particular style, unless you also add the some/full set of **place:XX** codes as `country` (or you set the min_zoom value to be better than 3.7).

Reply:
EG:

```
output_hstore['place:AR'] = 'country'
output_hstore['place:BD'] = 'country'
output_hstore['place:BR'] = 'country'
output_hstore['place:CN'] = 'country'
output_hstore['place:DE'] = 'country'
output_hstore['place:EG'] = 'country'
output_hstore['place:GB'] = 'country'
output_hstore['place:GR'] = 'country'
output_hstore['place:ID'] = 'country'
output_hstore['place:IL'] = 'country'
output_hstore['place:IN'] = 'country'
output_hstore['place:IT'] = 'country'
output_hstore['place:JP'] = 'country'
output_hstore['place:KO'] = 'country'
output_hstore['place:MA'] = 'country'
output_hstore['place:NL'] = 'country'
output_hstore['place:NP'] = 'country'
output_hstore['place:PK'] = 'country'
output_hstore['place:PL'] = 'country'
output_hstore['place:PS'] = 'country'
output_hstore['place:PT'] = 'country'
output_hstore['place:RU'] = 'country'
output_hstore['place:SA'] = 'country'
output_hstore['place:SE'] = 'country'
output_hstore['place:TR'] = 'country'
output_hstore['place:TW'] = 'country'
output_hstore['place:UA'] = 'country'
output_hstore['place:VN'] = 'country'
```

Reply:
Let's still do this now...

It's also related to work in https://github.com/tilezen/vector-datasource/issues/2077.

Reply:
Let's continue this in #2077... and reset to "place" = "country" here and for all 5.